x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/field_stats_factory.tsx

```diff
@@ -0,0 +1,327 @@
+    DataVisualizerPluginStart
+  >
+) => {
+  const [
+    { http, uiSettings, notifications, ...startServices },
+    { lens, data, usageCollection, fieldFormats },
+  ] = await getStartServices();
+
+  return {
+    http,
+    uiSettings,
+    data,
+    notifications,
+    lens,
+    usageCollection,
+    fieldFormats,
+    ...startServices,
+  };
+};
+
+export const getFieldStatsChartEmbeddableFactory = (
+  getStartServices: StartServicesAccessor<
+    DataVisualizerStartDependencies,
+    DataVisualizerPluginStart
+  >
+) => {
+  const factory: ReactEmbeddableFactory<
+    FieldStatisticsTableEmbeddableState,
+    FieldStatisticsTableEmbeddableState,
+    FieldStatisticsTableEmbeddableApi
+  > = {
+    type: FIELD_STATS_EMBEDDABLE_TYPE,
+    deserializeState: (state) => {
+      const serializedState = cloneDeep(state.rawState);
+      // inject the reference
+      const dataViewIdRef = state.references?.find(
+        (ref) => ref.name === FIELD_STATS_DATA_VIEW_REF_NAME
+      );
+      // if the serializedState already contains a dataViewId, we don't want to overwrite it. (Unsaved state can cause this)
+      if (dataViewIdRef && serializedState && !serializedState.dataViewId) {
+        serializedState.dataViewId = dataViewIdRef?.id;
+      }
+      return serializedState;
+    },
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const [coreStart, pluginStart] = await getStartServices();
+
+      const { http, uiSettings, notifications, ...startServices } = coreStart;
+      const { lens, data, usageCollection, fieldFormats } = pluginStart;
+
+      const deps = {
+        http,
+        uiSettings,
+        data,
+        notifications,
+        lens,
+        usageCollection,
+        fieldFormats,
+        ...startServices,
+      };
+      const {
+        api: timeRangeApi,
+        comparators: timeRangeComparators,
+        serialize: serializeTimeRange,
+      } = initializeTimeRange(state);
+
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+
+      const {
+        fieldStatsControlsApi,
+        dataLoadingApi,
+        fieldStatsControlsComparators,
+        serializeFieldStatsChartState,
+        onFieldStatsTableDestroy,
+        resetData$,
+      } = initializeFieldStatsControls(state);
+
+      const defaultDataViewId = await deps.data.dataViews.getDefaultId();
+      const dataViews$ = new BehaviorSubject<DataView[] | undefined>(
+        state.dataViewId !== '' && defaultDataViewId
+          ? [await deps.data.dataViews.get(state.dataViewId ?? defaultDataViewId)]
+          : undefined
+      );
+
+      const subscriptions = new Subscription();
+      subscriptions.add(
+        fieldStatsControlsApi.dataViewId$
+          .pipe(
+            skip(1),
+            skipWhile((dataViewId) => dataViewId === '' && defaultDataViewId === null),
+            switchMap((dataViewId) => deps.data.dataViews.get(dataViewId ?? defaultDataViewId))
```

Comment:
You need a try catch around `deps.data.dataViews.get`. This will throw if dataViewId is not found. Then publish error on `packages/presentation/presentation_publishing/interfaces/publishes_blocking_error.ts` interface.

Reply:
Updated here https://github.com/elastic/kibana/pull/184030/commits/f990d2278cc309c21b0893cd076b8e2f2ec127b1